File: backend-python/app/models/review.py
from dataclasses import dataclass, field
from datetime import datetime
from enum import StrEnum


class ReviewStatus(StrEnum):
    VISIBLE = "VISIBLE"
    HIDDEN = "HIDDEN"


@dataclass
class Review:
    review_id: str
    recipe_id: str
    user_id: str
    rating: int
    comment: str = ""
    status: ReviewStatus = ReviewStatus.VISIBLE
    reported_count: int = 0
    created_at: datetime = field(default_factory=datetime.now)
    updated_at: datetime = field(default_factory=datetime.now)

File: frontend/src/types/review.ts
/**
 * Review Types
 * レビュー型定義
 */

export interface Review {
  reviewId: string;
  recipeId: string;
  userId: string;
  rating: number;
  comment?: string;
  status: 'visible' | 'hidden';
  reportedCount: number;
  createdAt: string;
  updatedAt: string;
}

export interface CreateReviewRequest {
  rating: number;
  comment?: string;
}

export interface UpdateReviewRequest {
  rating: number;
  comment?: string;
}

export interface ReviewState {
  reviews: Review[];
  loading: boolean;
  error: string | null;
}
